src/box/arrow_ipc.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2024, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#include "arrow_ipc.h"
+
+#include "diag.h"
+#include "error.h"
+#include "tt_static.h"
+#include "nanoarrow/nanoarrow_ipc.h"
+
+/** Arrow IPC buffer, used for storing the encoded data. */
+struct arrow_ipc_buffer {
+	/** The implementation. */
+	struct ArrowBuffer base;
+};
+
+struct arrow_ipc_buffer *
+arrow_ipc_buffer_new(void)
+{
+	struct arrow_ipc_buffer *ipc_buffer = xmalloc(sizeof(*ipc_buffer));
+	ArrowBufferInit(&ipc_buffer->base);
+	return ipc_buffer;
+}
+
+void
+arrow_ipc_buffer_delete(struct arrow_ipc_buffer *ipc_buffer)
+{
+	struct ArrowBuffer *buffer = &ipc_buffer->base;
+	ArrowBufferReset(buffer);
+	TRASH(buffer);
+	free(buffer);
+}
+
+int
+arrow_ipc_encode(struct ArrowArray *array, struct ArrowSchema *schema,
```

Comment:
I don't see that this function is used anywhere or tested in CE. We shouldn't add dead code.

Reply:
Now it is used by `box_insert_arrow()`, which is not well tested in CE though…